src/main/java/org/folio/services/record/RecordServiceFactoryImpl.java

```diff
@@ -0,0 +1,24 @@
+package org.folio.services.record;
+
+import java.util.List;
+import java.util.Map;
+
+import org.folio.service.RecordService;
+import org.folio.service.RecordServiceImpl;
+import org.folio.service.spi.RecordServiceFactory;
+
+import io.vertx.core.Vertx;
+
+public class RecordServiceFactoryImpl implements RecordServiceFactory {
+
+  @Override
+  public RecordService create(Vertx vertx) {
+    return RecordServiceImpl.createForMultipleTables(
+      vertx,
+      Map.of(
+        "po_line",
+        List.of("po_line", "order_templates"),
+        "purchase_order",
+        List.of("purchase_order", "order_templates")));
```

Comment:
It looks like this implementation makes many calls in orders tables with this query: WHERE jsonb->'customFields' ? $1
In this case can you check if index was added for custom fields that will cover this query?
You can check example how to add indexes in schema.json.
So yo ujust need to update that file and RMB will handle index creating

Reply:
Good point. I'll check.

Reply:
Creating indexes using RMB seems unsuitable in this case.
Adding to `schema.json` for table `purchase_order`:
```
"ginIndex": [
        {
          "fieldName": "customFields",
          "caseSensitive": true,
          "removeAccents": false
        }
      ]
```
I was only able to create the following index: 
```
"purchase_order_customfields_idx_gin" gin ((jsonb ->> 'customFields'::text) gin_trgm_ops)
```

Which does not cover the query.

Instead I manually added a `gin ((jsonb -> 'customFields'::text))` index to every table effected by this PR - `purchase_order`, `po_line` and `order_templates`.

![explain_analyze](https://github.com/folio-org/mod-orders-storage/assets/25431636/24b031e1-c85a-421e-80f8-acc3fff926cd)

Reply:
The index creation worked with RMB, but the created index does not cover queries that use the `?` JSONB operator.

Reply:
can you also please ask this question in #raml-module-builder slack channel, if this is not supported, we will go with approach to manually create indexes as implemented now in the PR

Reply:
I can do that.